lib/sidekiq/web/application.rb

```diff
@@ -153,9 +157,15 @@ def self.set(key, val)
     end
 
     get "/morgue" do
-      @count = (params["count"] || 25).to_i
-      (@current_page, @total_size, @dead) = page("dead", params["page"], @count, reverse: true)
-      @dead = @dead.map { |msg, score| Sidekiq::SortedEntry.new(nil, score, msg) }
+      x = params[:substr]
+
+      if x && x != ""
+        @dead = search(Sidekiq::DeadSet.new, x)
+      else
+        @count = (params["count"] || 25).to_i
+        (@current_page, @total_size, @dead) = page("dead", params["page"], @count, reverse: true)
+        @dead = @dead.map { |msg, score| Sidekiq::SortedEntry.new(nil, score, msg) }
+      end
```

Comment:
This logic looks a little bit inconsistent, but I went for the least invasive change, so it should work exactly like before.

If I tested things correctly, these three filter routes boil down to filtering just the job ID, so usually there's just one result. I guess that's why there was no "real" pagination logic here? 🤔

Reply:
Pagination implies more state to remember across page loads which I wasn’t willing to build. If there are 1000 results, they’ll get a table with 1000 rows.

Reply:
I know very little about direct Redis interaction, but [`ZSCAN`](https://redis.io/docs/latest/commands/zscan/) seems to support passing a cursor argument, so it should be possible, right? (Possible follow-up PR material, hence the question.)

Reply:
Since the set is changing all the time, I didn’t want to implement cursor based filtering.